src/query/server/query_test.go

```diff
@@ -289,6 +291,45 @@ func TestIngestH2C(t *testing.T) {
 	testIngest(t, cfg, ctrl)
 }
 
+func TestPromRemoteBackend(t *testing.T) {
```

Comment:
We do, the test bodies are similar and can reuse some of the server startup code I introduced. I will do it in separate PR to not blow up this PR even more.

Reply:
Where? I only see `TestGRPCBackend`, nothing for `M3DBStorageType`.

Reply:
there are tests named `TestWriteH1` and `TestWriteH2C` that use configuration for m3db backend. However they mock out m3db client session so not sure how well they test m3db integration. I think it would be a good idea to at least add test case that runs M3DB in memory and writes/reads metric from it.